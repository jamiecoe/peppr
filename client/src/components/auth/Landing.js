import React, { Component } from 'react';
import { connect } from 'react-redux';
import { resetError } from '../../actions/auth';
import peppers from '../../assets/images/peppers.jpg';

import SignUp from './Signup';
import SignIn from './Signin';

export class Landing extends Component {
  componentDidMount() {
    this.props.resetError();
  }

  renderForm() {
    return this.props.match.path === '/' ? (
      <SignIn name="SignIn" />
    ) : (
      <SignUp name="SignUp" />
    );
  }

  render() {
    return (
      <div>
        <img
          className="landing__image"
          alt="red peppers on a dark background"
          src={peppers}
        />
        <div className="landing__container">
          <h1 className="landing__header">PEPPR</h1>
          <p className="landing__tagline">All your recipes in one place</p>
          {this.renderForm()}
        </div>
      </div>
    );
  }
}

export default connect(null, { resetError })(Landing);
