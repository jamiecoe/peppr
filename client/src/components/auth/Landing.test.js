import React from 'react';
import { shallow } from 'enzyme';
import { Landing } from './Landing';

describe('Landing', () => {
  const mockResetError = jest.fn();
  const props = {
    resetError: mockResetError,
    match: {
      path: '/',
    },
  };
  let landing = shallow(<Landing {...props} />, {
    disableLifecycleMethods: false,
  });

  it('should render properly, with <SignIn> component', () => {
    expect(landing).toMatchSnapshot();
    expect(landing.find('ReduxForm[name="SignIn"]').exists()).toBe(true);
    expect(landing.find('ReduxForm[name="SignUp"]').exists()).toBe(false);
  });

  it('should call `resetError` on componentDidMount()', () => {
    expect(mockResetError).toHaveBeenCalled();
  });

  describe('when props.match.path does not equal "/"', () => {
    beforeEach(() => {
      props.match.path = '/test';
      landing = shallow(<Landing {...props} />);
    });

    it('should render properly, with <SignOut> component', () => {
      expect(landing).toMatchSnapshot();
      expect(landing.find('ReduxForm[name="SignIn"]').exists()).toBe(false);
      expect(landing.find('ReduxForm[name="SignUp"]').exists()).toBe(true);
    });
  });
});
