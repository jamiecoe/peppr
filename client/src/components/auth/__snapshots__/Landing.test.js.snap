// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing should render properly, with <SignIn> component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Landing
    match={
        Object {
            "path": "/",
          }
    }
    resetError={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="red peppers on a dark background"
          className="landing__image"
          src="peppers.jpg"
/>,
        <div
          className="landing__container"
>
          <h1
                    className="landing__header"
          >
                    PEPPR
          </h1>
          <p
                    className="landing__tagline"
          >
                    All your recipes in one place
          </p>
          <ReduxForm
                    name="SignIn"
          />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "red peppers on a dark background",
          "className": "landing__image",
          "src": "peppers.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="landing__header"
>
              PEPPR
</h1>,
            <p
              className="landing__tagline"
>
              All your recipes in one place
</p>,
            <ReduxForm
              name="SignIn"
/>,
          ],
          "className": "landing__container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "PEPPR",
              "className": "landing__header",
            },
            "ref": null,
            "rendered": "PEPPR",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "All your recipes in one place",
              "className": "landing__tagline",
            },
            "ref": null,
            "rendered": "All your recipes in one place",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "SignIn",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="red peppers on a dark background"
            className="landing__image"
            src="peppers.jpg"
/>,
          <div
            className="landing__container"
>
            <h1
                        className="landing__header"
            >
                        PEPPR
            </h1>
            <p
                        className="landing__tagline"
            >
                        All your recipes in one place
            </p>
            <ReduxForm
                        name="SignIn"
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "red peppers on a dark background",
            "className": "landing__image",
            "src": "peppers.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="landing__header"
>
                PEPPR
</h1>,
              <p
                className="landing__tagline"
>
                All your recipes in one place
</p>,
              <ReduxForm
                name="SignIn"
/>,
            ],
            "className": "landing__container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "PEPPR",
                "className": "landing__header",
              },
              "ref": null,
              "rendered": "PEPPR",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "All your recipes in one place",
                "className": "landing__tagline",
              },
              "ref": null,
              "rendered": "All your recipes in one place",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "SignIn",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": false,
  },
}
`;

exports[`Landing when props.match.path does not equal "/" should render properly, with <SignOut> component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Landing
    match={
        Object {
            "path": "/test",
          }
    }
    resetError={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="red peppers on a dark background"
          className="landing__image"
          src="peppers.jpg"
/>,
        <div
          className="landing__container"
>
          <h1
                    className="landing__header"
          >
                    PEPPR
          </h1>
          <p
                    className="landing__tagline"
          >
                    All your recipes in one place
          </p>
          <ReduxForm
                    name="SignUp"
          />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "red peppers on a dark background",
          "className": "landing__image",
          "src": "peppers.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="landing__header"
>
              PEPPR
</h1>,
            <p
              className="landing__tagline"
>
              All your recipes in one place
</p>,
            <ReduxForm
              name="SignUp"
/>,
          ],
          "className": "landing__container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "PEPPR",
              "className": "landing__header",
            },
            "ref": null,
            "rendered": "PEPPR",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "All your recipes in one place",
              "className": "landing__tagline",
            },
            "ref": null,
            "rendered": "All your recipes in one place",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "SignUp",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="red peppers on a dark background"
            className="landing__image"
            src="peppers.jpg"
/>,
          <div
            className="landing__container"
>
            <h1
                        className="landing__header"
            >
                        PEPPR
            </h1>
            <p
                        className="landing__tagline"
            >
                        All your recipes in one place
            </p>
            <ReduxForm
                        name="SignUp"
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "red peppers on a dark background",
            "className": "landing__image",
            "src": "peppers.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="landing__header"
>
                PEPPR
</h1>,
              <p
                className="landing__tagline"
>
                All your recipes in one place
</p>,
              <ReduxForm
                name="SignUp"
/>,
            ],
            "className": "landing__container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "PEPPR",
                "className": "landing__header",
              },
              "ref": null,
              "rendered": "PEPPR",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "All your recipes in one place",
                "className": "landing__tagline",
              },
              "ref": null,
              "rendered": "All your recipes in one place",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "SignUp",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
